### PR TITLE
Update database schema to use boolean values where appropriate instead

### PR DESCRIPTION
**Pull request by Suhaibinator**
...of strings 'yes' and 'no'

Also update database sample and add a script to take a sample of the database.

### CODE REVIEW

**Comment by Suhaibinator** [comment]
This is the SQL code to be run on prod in order to make the changes.

```
ALTER TABLE Collections 
ADD COLUMN hasBooksBoolean BOOLEAN,
ADD COLUMN hasChaptersBoolean BOOLEAN,
ADD COLUMN hasVolumesBoolean BOOLEAN;

UPDATE Collections
SET hasBooksBoolean = (hasBooks = 'yes'),
    hasChaptersBoolean = (hasChapters = 'yes'),
    hasVolumesBoolean = (hasvolumes = 'yes');
```
    
The plan is to add the new columns for the API to use before we deprecate the old ones because of other apps that are dependent on the legacy fields. Once those apps are also transitioned, we can remove the old columns and rename the new ones.

File: models.py [hunk 1]
```diff
@@ -22,8 +22,8 @@ class HadithCollection(db.Model):
     def serialize(self):
         return {
             "name": self.name,
-            "hasBooks": self.hasbooks == "yes",
-            "hasChapters": self.haschapters == "yes",
+            "hasBooks": self.hasbooksBoolean,
```
Comment (hasankhan) [inline models.py:25]: Can we use a name without referring to type?

Some alternatives: includesBooks, containsBooks, booksExist

Comment (Suhaibinator) [reply models.py]: Done

**Comment by Suhaibinator** [comment]
Regarding commit hygiene, can we do a squash merge when this is approved and retain the necessary information in the squash-ed commit?

**Comment by hasankhan** [comment]
> Regarding commit hygiene, can we do a squash merge when this is approved and retain the necessary information in the squash-ed commit?

We want each change to have a separate commit in our git history. If we squash unrelated things into one commit, we pollute the history and make it difficult to understand what changed.

One pull request should change one thing at a time.

We can add the script for SQL dump in a separate pull request. 

When addressing feedback, please squash your commits locally  before sending it for review.

**Review by hasankhan — APPROVED** [review]
Lgtm

File: docker-compose.yml [hunk 2]
```diff
@@ -8,6 +8,8 @@ services:
       - ./db:/docker-entrypoint-initdb.d/:ro
     env_file:
       - .env.local
+    ports:
```
Comment (hasankhan) [inline docker-compose.yml:11]: Please remove this from this pull request

File: db/01-hadithTable.sql
Comment (hasankhan) [inline db/01-hadithTable.sql:1]: Why is this file removed?
